lib/portlayer/util/util.go

```diff
@@ -61,30 +62,49 @@ func ServiceURL(serviceName string) *url.URL {
 }
 
 // Update the VM display name on vSphere UI
-func DisplayName(config *spec.VirtualMachineConfigSpecConfig, namingConvention string) string {
-	var name string
-	shortID := config.ID[:constants.ShortIDLen]
-	nameMaxLen := constants.MaxVMNameLength - len(shortID)
-	prettyName := config.Name
-
-	if namingConvention != "" {
-		//TODO: need to respect max length -- potentially enforce '-' separator
-		if strings.Contains(namingConvention, "{id}") {
-			name = strings.Replace(namingConvention, "{id}", shortID, -1)
-		} else {
-			//assume name
-			name = strings.Replace(namingConvention, "{name}", prettyName, -1)
+func DisplayName(cfg *spec.VirtualMachineConfigSpecConfig, namingConvention string) string {
+	shortID := cfg.ID[:constants.ShortIDLen]
+	prettyName := cfg.Name
+
+	if namingConvention == "" {
+		namingConvention = config.DefaultNamePattern
+	}
+	name := namingConvention
+
+	// determine length of template without tokens
+	templateLen := len(strings.Replace(strings.Replace(namingConvention, config.NameToken.String(), "", -1), config.IDToken.String(), "", -1))
+	availableLen := constants.MaxVMNameLength - templateLen
+
+	if availableLen < 0 {
```

Comment:
Should we (additionally) have validation of the length of the template when the CNC is set? If the user enters a template that will hit this case, we should at least warn then. (And in the case of the UI we should probably be reflecting this limit and the truncation logic in the generated examples.)